civis/ml/_model.py

```diff
@@ -140,9 +150,12 @@ def _load_estimator(job_id, run_id, filename='estimator.pkl', client=None):
     if not HAS_JOBLIB:
         raise ImportError('Install joblib to download models '
                           'from Civis Platform.')
-    with tempfile.TemporaryDirectory() as tempdir:
-        path = _retrieve_file(filename, job_id, run_id, tempdir, client=client)
-        return joblib.load(path)
+    tempdir = tempfile.mkdtemp()
+    path = _retrieve_file(filename, job_id, run_id, tempdir, client=client)
+    obj = joblib.load(path)
+    os.remove(path)
+    os.removedirs(tempdir)
```

Comment:
The cleanup should be in a `finally` block. Maybe `shutil.rmtree` would be better here?